src/visitors.rs

```diff
@@ -144,8 +176,7 @@ impl<'a, 'b: 'a, 'tcx: 'b> MirVisitor<'a, 'b, 'tcx> {
         }
         // The entry block has no predecessors and its initial state is the function parameters
         // as well any promoted constants.
-        let state_with_parameters = Environment::with_parameters(self.mir.arg_count);
```

Comment:
What happened to self.mir.arg_count? Do you still need that?

Reply:
I still need self.mir. The arg_count I don't need, but mir is not my data structure.